src/pw/gpu/pw_gpu_internal.cc

```diff
@@ -0,0 +1,550 @@
+/*----------------------------------------------------------------------------*/
+/*  CP2K: A general program to perform molecular dynamics simulations         */
+/*  Copyright 2000-2022 CP2K developers group <https://cp2k.org>              */
+/*                                                                            */
+/*  SPDX-License-Identifier: GPL-2.0-or-later                                 */
+/*----------------------------------------------------------------------------*/
+
+#include <array>
+#include <map>
+#include <string>
+#include <unistd.h>
+#include <vector>
+
+#include "../../offload/offload_library.h"
+
+#ifdef __PW_CUDA
+#include "cuda/cuda_fft_private_header.hpp"
+#endif
+
+#ifdef __PW_HIP
+#include "hip/hip_fft_private_header.hpp"
+#endif
+
+void gpu_scatter(offloadStream_t &stream__, const double scale__, int num_points__,
+                 const int nmaps__, const int *map_index__,
+                 const double *dataIn__, double *DataOut__);
+void gpu_gather(offloadStream_t &stream__, const double scale__, int num_points__,
+                const int *map_index__, const double *dataIn__,
+                double *DataOut__);
+
+static bool is_configured{false};
+
+static blasHandle_t handle_;
+static offloadStream_t streams_;
+
+// INIT/RELEASE
+extern "C" int pw_gpu_init() {
+  if (!is_configured) {
+    offload_set_device();
+    /* stream allocation  */
+    CreateStream(streams_);
+    blasCreate(handle_);
+    blasSetStream(handle_, streams_);
+    is_configured = true;
+  }
+  printf("hip fft : Initialization done!!!");
```

Comment:
It might be that this print statement interferes with the communication between the cp2k shell and the do_regtest.py script.

Reply:
thanks for spotting it.

Reply:
Oh I use the offload interface to the GPUs functions now.